src/tools/gophish_export.py

```diff
@@ -253,6 +253,88 @@ def get_application(rawEvent):
     return application
 
 
+def find_unique_target_clicks_count(clicks):
+    """Return the number of unique clicks in a click dict."""
+    uniq_users = set()
+    for click in clicks:
+        uniq_users.add(click["user"])
+    return len(uniq_users)
+
+
+def write_campaign_summary(api, assessment_id):
+    """Output a campaign summary report to JSON, console, and a text file."""
+    campaign_ids = get_campaign_ids(api, assessment_id)
+    campaign_data_template = "campaign_data.json"
+    campaign_summary_json = f"{assessment_id}_campaign_data.json"
+    campaign_summary_textfile = f"{assessment_id}_summary_{datetime.strftime(datetime.now(), '%Y-%m-%dT%H:%M:%S')}.txt"
+
+    with open(campaign_data_template) as template:
+        campaign_data = json.load(template)
+
+    logging.info("Writing campaign summary report to %s", campaign_summary_textfile)
+    file_out = open(campaign_summary_textfile, "w+")
+    file_out.write("Campaigns for Assessment: " + assessment_id)
+
+    for campaign_id in campaign_ids:
+        campaign = api.campaigns.get(campaign_id)
+        if campaign.name.endswith("_level-1"):
+            level = "level-1"
+        elif campaign.name.endswith("_level-2"):
+            level = "level-2"
+        elif campaign.name.endswith("_level-3"):
+            level = "level-3"
+        elif campaign.name.endswith("_level-4"):
+            level = "level-4"
+        elif campaign.name.endswith("_level-5"):
+            level = "level-5"
+        elif campaign.name.endswith("_level-6"):
+            level = "level-6"
+        else:
```

Comment:
While you're touching this code anyway, this logic could me made less repetitive and less error-prone if other levels are ever added by using a regex.  I'm thinking of something like this:
 ```suggestion
    regex = re.compile(r"^.*_(?P<level>level-[1-6])$")
    for campaign_id in campaign_ids:
        campaign = api.campaigns.get(campaign_id)
        match = regex.fullmatch(campaign.name)
        if match:
          level = match.group("level")
        else:
```

Reply:
See 5816a24f37e9a600f2b55946de18b7acb8d0da07 for changes to this file